CRM/Civiquickbooks/Form/Settings.php

```diff
@@ -139,10 +139,22 @@ public function getFormSettings() {
   public function saveSettings() {
     $settings = $this->getFormSettings();
     $values = array_intersect_key($this->_submittedValues, $settings);
+
+    // Fix for unsetting a checkbox. When setting a checkbox,
+    // quickbooks_bool_value => 1 is returned. But when unsetting a checkbox,
+    // quickbooks_bool_value => 0 is NOT returned. Absence of a checkbox
+    // attribute should be interpreted as setting its value to 0.
+    foreach ($settings as $setting_name => $setting_info) {
+        if ($setting_info['type'] == "Boolean" && !array_key_exists($setting_name, $values)) {
+            $values[$setting_name] = 0;
+        }
+    }
+
     $previousValues = CRM_Quickbooks_APIHelper::getQuickBooksCredentials();
 
     civicrm_api3('setting', 'create', $values);
 
+    // FIXME this erases valid access tokens any time settings are saved
```

Comment:
I want to fix this but want to understand what the scenario is that this if statement is allowing for.